VerticalSlice2D/Assets/Scripts/TPBar.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class TPBar : MonoBehaviour
+{
+    private float startingY;
+    [SerializeField] private TextMeshProUGUI tpText;
+    [SerializeField] private GameObject orangeTPBar;
+    [SerializeField] private GameObject max;
+    [SerializeField] private GameObject tpPercentage;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        Transform spriteMask = GameObject.Find("TPBar_Spritemask").GetComponent<Transform>();
+        startingY = spriteMask.position.y;
+    }
+
+    void UpdateSprite()
+    {
+        GameObject thePlayer = GameObject.Find("Player");
```

Comment:
Alles wat je gebruikt van speler in een variabele opslaan. Dat zorgt ervoor dat player niet gezocht hoeft te worden elke frame.